## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 453610`
Commit: [`149d20db37c2aacfdd6c0bdb07c8e8412ff92032`](https://github.com/NixOS/nixpkgs/commit/149d20db37c2aacfdd6c0bdb07c8e8412ff92032) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/149d20db37c2aacfdd6c0bdb07c8e8412ff92032..pull/453610/head))
Merge: [`f2148450e169177b414352b516fcf978d862ebc8`](https://github.com/NixOS/nixpkgs/commit/f2148450e169177b414352b516fcf978d862ebc8)

Logs: https://github.com/qbisi/nixpkgs-review-gha/actions/runs/18636557671


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 5 packages built:</summary>
  <ul>
    <li>precice</li>
    <li>python312Packages.pyprecice</li>
    <li>python312Packages.pyprecice.dist</li>
    <li>python313Packages.pyprecice</li>
    <li>python313Packages.pyprecice.dist</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:x: 5 packages failed to build:</summary>
  <ul>
    <li>precice</li>
    <li>python312Packages.pyprecice</li>
    <li>python312Packages.pyprecice.dist</li>
    <li>python313Packages.pyprecice</li>
    <li>python313Packages.pyprecice.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-linux`</summary>
<details>
<summary>precice</summary>
<pre>size1            =  87.51 sec*proc (556 tests)
size2            = 157.35 sec*proc (333 tests)
size3            =  41.54 sec*proc (75 tests)
size4            = 150.15 sec*proc (231 tests)
size5            =   0.73 sec*proc (1 test)
sockets          =   5.08 sec*proc (10 tests)
solverdummy      =   8.49 sec*proc (5 tests)
testing          =   2.58 sec*proc (9 tests)
time             =   2.23 sec*proc (16 tests)
tools            =   4.60 sec*proc (30 tests)
utils            =  10.67 sec*proc (55 tests)
valid            =   1.05 sec*proc (7 tests)
xml              =   1.69 sec*proc (10 tests)

Total Test time (real) = 113.04 sec

The following tests FAILED:
	851 - precice.Integration/Serial/Time/Explicit/ParallelCoupling/ReadWriteScalarDataWithSubcycling6400Steps (Failed) integration size2
Errors while running CTest
make: *** [Makefile:91: test] Error 8</pre>
</details>
</details>

---
### `x86_64-darwin`
:white_check_mark: *No rebuilds*

---
### `aarch64-darwin` (sandbox = true)
<details>
  <summary>:white_check_mark: 5 packages built:</summary>
  <ul>
    <li>precice</li>
    <li>python312Packages.pyprecice</li>
    <li>python312Packages.pyprecice.dist</li>
    <li>python313Packages.pyprecice</li>
    <li>python313Packages.pyprecice.dist</li>
  </ul>
</details>

--- a/pkgs/by-name/pr/precice/package.nix
+++ b/pkgs/by-name/pr/precice/package.nix
@@ -3,29 +3,32 @@
   stdenv,
   fetchFromGitHub,
   cmake,
+  pkg-config,
   boost,
   eigen,
   libxml2,
   mpi,
   python3Packages,
   petsc,
-  pkg-config,
+  ctestCheckHook,
+  mpiCheckPhaseHook,
 }:
 
-stdenv.mkDerivation {
+assert petsc.mpiSupport;
+
+stdenv.mkDerivation (finalAttrs: {
   pname = "precice";
-  version = "3.2.0-unstable-2025-05-23";
+  version = "3.3.0";
 
   src = fetchFromGitHub {
     owner = "precice";
     repo = "precice";
-    rev = "6ee3e347843d4d3c416a32917f6505d35b822445";
-    hash = "sha256-BxNAbpeLqJPzQ9dvvgC9jJQQFBdVMunSqIekz7SIHv4=";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-1FbTNo2F+jH1EVV6gXc9o0T31UHY/wBK3vQeCV7wW5E=";
   };
 
   cmakeFlags = [
-    (lib.cmakeBool "PRECICE_PETScMapping" false)
-    (lib.cmakeBool "BUILD_SHARED_LIBS" true)
+    (lib.cmakeBool "BUILD_SHARED_LIBS" (!stdenv.hostPlatform.isStatic))
   ];
 
   nativeBuildInputs = [
@@ -43,12 +46,26 @@ stdenv.mkDerivation {
     python3Packages.numpy
   ];
 
+  __darwinAllowLocalNetworking = true;
+
+  doCheck = true;
+
+  nativeCheckInputs = [
+    ctestCheckHook
+    mpiCheckPhaseHook
+  ];
+
+  disabledTests = [
+    # Because preciceDt becomes very small. Test is likely to fail on other platform.
+    "precice.Integration/Serial/Time/Explicit/ParallelCoupling/ReadWriteScalarDataWithSubcycling6400Steps"
+  ];
+
   meta = {
     description = "PreCICE stands for Precise Code Interaction Coupling Environment";
     homepage = "https://precice.org/";
-    license = with lib.licenses; [ gpl3 ];
+    license = with lib.licenses; [ lgpl3Only ];
     maintainers = with lib.maintainers; [ Scriptkiddi ];
-    mainProgram = "binprecice";
+    mainProgram = "precice-tools";
     platforms = lib.platforms.unix;
   };
-}
+})

--- a/pkgs/development/python-modules/pyprecice/default.nix
+++ b/pkgs/development/python-modules/pyprecice/default.nix
@@ -5,9 +5,9 @@
 
   # build-system
   cython,
-  pip,
   pkgconfig,
   setuptools,
+  setuptools-git-versioning,
 
   # dependencies
   mpi4py,
@@ -17,38 +17,38 @@
 
 buildPythonPackage rec {
   pname = "pyprecice";
-  version = "3.2.1";
+  version = "3.3.1";
   pyproject = true;
 
   src = fetchFromGitHub {
     owner = "precice";
     repo = "python-bindings";
     tag = "v${version}";
-    hash = "sha256-8AM2wbPX54UaMO4MzLOV0TljLTAPOqR9gUbtT2McNjs=";
+    hash = "sha256-NkTrMZ7UKB5O2jIlhLhgkOm8ZeWJA1FoursA1df7XOk=";
   };
-
-  postPatch = ''
-    substituteInPlace pyproject.toml \
-      --replace-fail "setuptools>=61,<72" "setuptools"
-  '';
 
   build-system = [
     cython
-    pip
     pkgconfig
     setuptools
+    setuptools-git-versioning
   ];
 
   dependencies = [
     numpy
     mpi4py
+  ];
+
+  buildInputs = [
     precice
   ];
 
-  # Disable Test because everything depends on open mpi which requires network
+  # no official test instruction
   doCheck = false;
 
-  # Do not use pythonImportsCheck because this will also initialize mpi which requires a network interface
+  pythonImportsCheck = [
+    "precice"
+  ];
 
   meta = {
     description = "Python language bindings for preCICE";

